packages/x-data-grid-premium/src/hooks/features/rowGrouping/createGroupingColDef.tsx

```diff
@@ -12,10 +12,13 @@ import { GridColumnRawLookup, isSingleSelectColDef } from '@mui/x-data-grid-pro/
 import { GridApiPremium } from '../../../models/gridApiPremium';
 import { GridGroupingColumnFooterCell } from '../../../components/GridGroupingColumnFooterCell';
 import { GridGroupingCriteriaCell } from '../../../components/GridGroupingCriteriaCell';
+import { GridDataSourceGroupingCriteriaCell } from '../../../components/GridDataSourceGroupingCriteriaCell';
 import { GridGroupingColumnLeafCell } from '../../../components/GridGroupingColumnLeafCell';
 import {
   getRowGroupingFieldFromGroupingCriteria,
   GRID_ROW_GROUPING_SINGLE_GROUPING_FIELD,
+  ROW_GROUPING_STRATEGY,
+  DATA_SOURCE_ROW_GROUPING_STRATEGY,
```

Comment:
`ROW_GROUPING_STRATEGY_DATA_SOURCE` would be easier to read.

Reply:
I tried to align the [one](https://github.com/mui/mui-x/blob/153bce3cd50df281801ab8a4721d3aa43a919023/packages/x-data-grid-pro/src/hooks/features/serverSideTreeData/useGridDataSourceTreeDataPreProcessors.tsx#L38) with the tree data. Do you think we should rename both?
Tbh, the current one looks more readable to me personally, but I'm fine to rename if it looks less readable to other too. 👍

Reply:
1.
```
  ROW_GROUPING_STRATEGY,
  DATA_SOURCE_ROW_GROUPING_STRATEGY,
```

2.
```
  ROW_GROUPING_STRATEGY,
  ROW_GROUPING_STRATEGY_DATA_SOURCE,
```

I find option 2 more readable when constants are used like enum values, imho it makes it apparent that both values have the same "type".

Reply:
And tbh I would actually rename to something like this:

```
  ROW_GROUPING_STRATEGY_DEFAULT,
  ROW_GROUPING_STRATEGY_DATA_SOURCE,
```